CMakeLists.txt

```diff
@@ -59,47 +59,51 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1y")
 # external libraries and definitions for those external libraries.
 macro(add_phobos_executable target_name)
     add_chibios_executable(${target_name} ${ARGN} ${PHOBOS_COMMON_SRC})
-    set(source_file_properties_modified 0)
+
+    set(USE_BICYCLE_SOURCE FALSE)
     foreach(src in ${ARGN})
         get_filename_component(src_name ${src} NAME)
+
+        # Determine if bicycle sources are included in this project
+        if(src_name STREQUAL "bicycle.cc")
+            set(USE_BICYCLE_SOURCE TRUE)
+        endif()
+    endforeach()
+
+    foreach(src in ${ARGN})
+        get_filename_component(src_name ${src} NAME)
+
+        # suppress Boost undef warnings
+        if(USE_BICYCLE_BOOST_SUBMODULE AND
+           src_name MATCHES "(main|receiver|txrx_common).cc")
```

Comment:
`receiver.cc` and `txrx_common.cc` shouldn't be using anything from Boost. Perhaps `type_traits` is getting pulled in from Boost instead of the standard library?

Reply:
`inc/utility.h` uses `boost::math` which pulls in `boost::type_traits`.